File: core/MaximumLengthOfRepeatedSubarray/MaximumLengthOfRepeatedSubarray.java
package com.dbc;

import java.util.Arrays;

public class MaximumLengthOfRepeatedSubarray {
    public int findLength(int[] nums1, int[] nums2) {
        int[] dp = new int[nums2.length + 1];
        Arrays.fill(dp, 0);
        int res = 0;
        for (int i = 1; i < nums1.length + 1; i++) {
            int remain = 0;
            for (int j = 1; j < nums2.length + 1; j++) {
                int temp = dp[j];
                if (nums1[i - 1] == nums2[j - 1]) dp[j] = remain + 1;
                else dp[j] = 0;
                remain = temp;
                res = Math.max(res, dp[j]);
            }
        }
        return res;
    }
}

File: core/MaximumLengthOfRepeatedSubarray/MaximumLengthOfRepeatedSubarray.py
from typing import List


def findLength(nums1: List[int], nums2: List[int]) -> int:
    rows, cols = len(nums1) + 1, len(nums2) + 1
    dp, res = [0] * cols, 0
    for row in range(1, rows):
        remain = 0
        for col in range(1, cols):
            temp = dp[col]
            if nums1[row - 1] == nums2[col - 1]:
                dp[col] = remain + 1
            else:
                dp[col] = 0
            remain = temp
            res = max(res, dp[col])

    return res


if __name__ == '__main__':
    print(findLength([1, 0, 0, 0, 1], [1, 0, 0, 1, 1]))
